### PR TITLE
#89377 After following survey, offered to discard cards

### PR DESCRIPTION
**Pull request by xmakina**
Proceed to next when active player isn't leader in extra turn

### CODE REVIEW

**Review by elaskavaia — COMMENTED** [review]
I don't think you can get getActivatePlayerId here it may not be active player state,
why you cannot use $player_id from this function? Is it the player in question?

File: eminentdomain.game.php [hunk 1]
```diff
@@ -1241,6 +1241,13 @@ function next_RoleDispatch($player_id) {
                     return; // cannot go pre-discard yet
                 }
             }
+
+            $leader = $this->getGameStateValue('leader');
+            if ($leader != $this->getActivePlayerId()){ // Follow masquerading as Turn e.g. Survey
```
Comment (elaskavaia) [inline eminentdomain.game.php]: why not $player_id?

Comment (xmakina) [reply eminentdomain.game.php]: Switched to `player_id`

File: eminentdomain.game.php [hunk 2]
```diff
@@ -1241,6 +1241,13 @@ function next_RoleDispatch($player_id) {
                     return; // cannot go pre-discard yet
                 }
             }
+
+            $leader = $this->getGameStateValue('leader');
```
Comment (elaskavaia) [inline eminentdomain.game.php:1245]: I will change to
``
            $leader = $this->getGameStateValue('leader');
            if ($leader == $player_id && $this->getUntrivialDiscardOrActivate($color) && !$discardPlayed && $rolePlayed >= $roleAllowerd ) {
                $this->gamestate->nextState('discard'); // pre-discard
                return;
            }

**Comment by elaskavaia** [comment]
I merged similar fix

**Pull request closed without merging**